src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -529,7 +534,7 @@ public OpenIdConnectAuthenticationHandler(HttpClient backchannel)
             if (subjectClaimType == null)
             {
                 Logger.LogError(string.Format(CultureInfo.InvariantCulture, Resources.OIDCH_0041_Subject_Claim_Not_Found, identity.ToString()));
-                return ticket;
+                return null;
```

Comment:
No, please, stop throwing exceptions everywhere... or catch them eventually to avoid returning a 500 response (I'll open a ticket for the OAuth2 bearer middleware as it's really an annoying default behavior).

I'm not sure to understand why logging the error and returning `null` wouldn't be enough.